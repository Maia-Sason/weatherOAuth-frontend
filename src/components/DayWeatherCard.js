import { Paper, Box, Typography, SvgIcon, makeStyles } from "@material-ui/core";
import WeatherIcon from "./WeatherIcon";

const useStyles = makeStyles((theme) => ({
  paper: {
    background: "none",
    height: "100px",
    margin: "5px",
    flexDirection: "row",
    alignItems: "center",
    color: "white",
  },
  box: {
    padding: "1em",
    display: "flex",
    justifyContent: "space-between",
  },
  icon: {
    height: "3em",
    width: "3em",
    fill: "white",
  },
  text: {
    fontSize: "2em",
  },
  day: {
    paddingTop: 5,
    paddingLeft: 3,
  },
}));

const DayWeatherCard = ({ icon, temp, day }) => {
  const classes = useStyles();
  const options = { weekday: "long" };

  const weekDay = new Intl.DateTimeFormat("en-US", options).format(day);
  return (
    <>
      <Paper className={classes.paper}>
        <Typography className={classes.day}>{weekDay}</Typography>
        <Box className={classes.box}>
          <Typography className={classes.text}>{temp}&deg;</Typography>
          <Box className={classes.icon}>
            <WeatherIcon icon={icon}></WeatherIcon>
          </Box>
        </Box>
      </Paper>
    </>
  );
};

export default DayWeatherCard;
