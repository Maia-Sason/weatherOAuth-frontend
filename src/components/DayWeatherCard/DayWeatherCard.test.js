import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import DayWeatherCard from "./DayWeatherCard";

test("Matches snapshot", () => {
  const component = render(<DayWeatherCard />);
  expect(component.container).toMatchSnapshot();
});

it("Renders DayWeatherCard without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<DayWeatherCard />, div);
});
