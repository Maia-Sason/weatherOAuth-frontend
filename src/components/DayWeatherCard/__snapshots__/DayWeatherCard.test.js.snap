// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation10 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-6 makeStyles-box-2"
    >
      <p
        class="MuiTypography-root makeStyles-text-4 MuiTypography-body1"
      >
        Â°
      </p>
      <div
        class="MuiBox-root MuiBox-root-7 makeStyles-icon-3"
      />
    </div>
    <p
      class="MuiTypography-root makeStyles-day-5 MuiTypography-body1"
    >
      Friday
    </p>
  </div>
</div>
`;
